src/fsevent.rs

```diff
@@ -450,13 +444,9 @@ impl FsEventWatcher {
                 fs::FSEventStreamInvalidate(stream);
                 fs::FSEventStreamRelease(stream);
             }
-
-            done_tx
-                .send(())
-                .expect("error while signal run loop is done");
         });
         // block until runloop has been sent
-        self.runloop = Some((rl_rx.recv().unwrap().0, done_rx));
+        self.runloop = Some((rl_rx.recv().unwrap().0, thread_handle));
```

Comment:
Is there a chance this might fail / get poisoned ? Because then I'd like to not just panic and use the Error we can return.

Reply:
I think the only place where the thread will panic would be when we're trying to send the runloop back to the main thread. I don't believe that can happen here, as best as I can tell. Furthermore, the old behavior would have also panicked in `stop()` if we didn't receive confirmation we shut down the thread.

We could update this to propagate the PoisonError though. Would you want this converted into error::Error::Generic, or to add a new error variant?

Reply:
I think error::Error::Generic is totally fine. I'd just try to have less code/crates that can panic your application instead of catching such errors at points the developer can decide.

Reply:
ping: do you want to resolve this or should I go on, merge it and issue a release so it can be fixed later ?

Reply:
Hello! Sorry, got distracted. Anyway, looking back through the code there's no place where the code could panic in our code. So the only place this that could happen would be if the fsevents crate (which is unlikely since we're just making direct calls into the C library) or std::thread library panicked. Since we can't be assured what internal invariants might have been violated if that happens, I think panicking is the appropriate course of action.